zoltar-core/src/main/java/com/spotify/zoltar/ModelLoader.java

```diff
@@ -48,19 +49,141 @@
   }
 
   /**
-   * Lifts a supplier into a {@link ModelLoader}.
+   * PreLoader is model loader that calls {@link ModelLoader#get()} allowing model preloading.
+   *
+   * @param <M> Model instance type.
+   */
+  @FunctionalInterface
+  interface PreLoader<M extends Model<?>> extends ModelLoader<M> {
+
+    /**
+     * Returns a blocking {@link ModelLoader}. Blocks till the model is loaded or a {@link Duration}
+     * is met.
+     *
+     * @param supplier model supplier.
+     * @param duration Amount of time that it should wait, if necessary, for model to be loaded.
+     * @param executor the executor to use for asynchronous execution.
+     * @param <M>      Underlying model instance.
+     */
+    static <M extends Model<?>> PreLoader<M> preload(final ThrowableSupplier<M> supplier,
+                                                     final Duration duration,
+                                                     final Executor executor)
+        throws InterruptedException, ExecutionException, TimeoutException {
+      return preload(ModelLoader.load(supplier, executor), duration)::get;
+    }
+
+    /**
+     * Returns a blocking {@link PreLoader}. Blocks till the model is loaded or a {@link Duration}
+     * is met.
+     *
+     * @param loader model loader.
+     * @param duration Amount of time that it should wait, if necessary, for model to be loaded.
+     * @param <M>    Underlying model instance.
+     */
+    static <M extends Model<?>> PreLoader<M> preload(final ModelLoader<M> loader,
+                                                     final Duration duration)
+        throws InterruptedException, ExecutionException, TimeoutException {
+      return ModelLoader.loaded(loader.get(duration))::get;
+    }
+
+    /**
+     * Returns a blocking {@link PreLoader}. Blocks till the model is loaded or a {@link Duration}
+     * is met.
+     *
+     * @param duration Amount of time that it should wait, if necessary, for model to be loaded.
+     */
+    static <M extends Model<?>> Function<ModelLoader<M>, PreLoader<M>> preload(
+        final Duration duration) {
+      return loader -> {
+        try {
+          return preload(loader, duration);
+        } catch (final Exception e) {
+          final CompletableFuture<M> failed = new CompletableFuture<>();
+          failed.completeExceptionally(e);
+
+          return () -> failed;
+        }
+      };
+    }
+
+  }
+
+  /**
+   * ConsLoader is a constant {@link ModelLoader}.
+   *
+   * @param <M> Model instance type.
+   */
+  @FunctionalInterface
+  interface ConsLoader<M extends Model<?>> extends ModelLoader<M> {
```

Comment:
Why not ``ConstantLoader`? ditto for the `cons` function, why not `constant`?

Reply:
functional lingo

Reply:
Lisp?

Reply:
yeah and you can see it in haskell and scala as well.